cmd/earth/main.go

```diff
@@ -1237,6 +1323,249 @@ func (app *earthApp) actionRegister(c *cli.Context) error {
+		return errors.Wrap(err, "failed to remove account tokens")
+	}
+	return nil
+}
+
+func (app *earthApp) actionAccountLogin(c *cli.Context) error {
+	app.commandName = "actionAccountLogin"
+	email := app.email
+	token := app.token
+	pass := app.password
+
+	if c.NArg() == 1 {
+		emailOrToken := c.Args().First()
+		if token == "" && email == "" {
+			if secretsclient.IsValidEmail(emailOrToken) {
+				email = emailOrToken
+			} else {
+				token = emailOrToken
+			}
+
+		} else {
+			return errors.New("invalid number of arguments provided")
+		}
+	} else if c.NArg() > 1 {
+		return errors.New("invalid number of arguments provided")
+	}
+
+	if token != "" && (email != "" || pass != "") {
+		return errors.New("--token can not be used in conjuction with --email or --password")
+	}
+	sc, err := secretsclient.NewClient(app.apiServer, app.sshAuthSock, app.console.Warnf)
+	if err != nil {
+		return err
+	}
+
+	if token != "" || pass != "" {
+		err := sc.DeleteCachedCredentials()
+		if err != nil {
+			return errors.Wrap(err, "failed to clear cached credentials")
+		}
+		sc.DisableSSHKeyGuessing()
+	} else if email != "" {
+		foundSSHKeys, err := sc.FindSSHAuth()
+		if err == nil {
+			if keys, ok := foundSSHKeys[email]; ok {
+				if len(keys) > 0 {
+					foundSSHKey := keys[0]
+					err := sc.SetLoginSSH(email, foundSSHKey)
+					if err != nil {
+						return err
+					}
+					fmt.Printf("Logged in as %q using ssh auth\n", email)
+					return nil
+				}
+			}
+		}
+	}
+
+	loggedInEmail, authType, err := sc.WhoAmI()
+	switch errors.Cause(err) {
+	case secretsclient.ErrNoAuthorizedPublicKeys, secretsclient.ErrNoSSHAgent:
+		break
+	case nil:
+		if email != "" && email != loggedInEmail {
+			break // case where a user has multiple emails and wants to switch
+		}
+		fmt.Printf("Logged in as %q using %s auth\n", loggedInEmail, authType)
+		return nil
+	default:
+		return err
+	}
+
+	if email == "" && token == "" {
+		emailOrToken := promptInput("enter your email or auth token: ")
+		if strings.Contains(emailOrToken, "@") {
+			email = emailOrToken
+		} else {
+			token = emailOrToken
+		}
+	}
+
+	if email != "" && pass == "" {
+		passwordBytes, err := password.Read("enter your password: ")
+		if err != nil {
+			return err
+		}
+		pass = string(passwordBytes)
+		if pass == "" {
+			return fmt.Errorf("no password entered")
+		}
+	}
+
+	if token != "" {
+		email, err = sc.SetLoginToken(token)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Logged in as %q using token auth\n", email) // TODO display if using read-only token
+	} else {
+		err = sc.SetLoginCredentials(email, string(pass))
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Logged in as %q using password auth\n", email)
+		fmt.Printf("Warning unencrypted password has been stored under ~/.earthly/auth.token; consider using ssh-based auth to prevent this.\n")
+	}
+	return nil
+}
+
+func (app *earthApp) actionAccountLogout(c *cli.Context) error {
+	app.commandName = "actionAccountLogout"
```

Comment:
```suggestion
	app.commandName = "accountLogout"
```

(consistency)

Reply:
Can you also double check that these are correct in all other places? Don't know if I missed any.

Reply:
thanks for pointing that out, found a few bad copy/paste places that caused this.